app/check_tx.go

```diff
@@ -15,6 +15,15 @@ import (
 // transactions that contain blobs.
 func (app *App) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
 	tx := req.Tx
+
+	// all txs should be under the max tx size limit
```

Comment:
we can extract this logic into a helper to make it less repetitive once we decide what to do with the AnteHandler because otherwise it becomes confusing.